larray/inout/common.py

```diff
@@ -92,13 +93,12 @@ def close(self):
         raise NotImplementedError()
 
     def _get_original_file_name(self):
-        if self.overwrite_file and os.path.isfile(self.fname):
+        if self.overwrite_file and self.fname.is_file():
             self.original_file_name = self.fname
-            fname, ext = os.path.splitext(self.fname)
-            self.fname = f'{fname}~{ext}'
+            self.fname = self.fname.parent / f'{self.fname.stem}~{self.fname.suffix}'
```

Comment:
self.fname = self.fname.with_stem(self.fname.stem + '~')
?

Reply:
yes

Reply:
``with_stem`` only exists in Python 3.9+

Reply:
oups, sorry